ClickToCall.Web/Services/RequestValidationService.cs

```diff
@@ -10,16 +13,46 @@ public interface IRequestValidationService
 
     public class RequestValidationService : IRequestValidationService
     {
-        private readonly RequestValidator _requestValidator;
-
-        public RequestValidationService()
-        {
-            _requestValidator = new RequestValidator();
-        }
-
         public bool IsValidRequest(HttpContext context, string authToken)
         {
-            return _requestValidator.IsValidRequest(context, authToken);
+            // https://github.com/twilio/twilio-csharp/blob/ab0983ae00269746c5dbeb891955ae3edd83796d/src/Twilio.Twiml/RequestValidator.cs
+
+            if (context.Request.IsLocal)
+            {
+                return true;
+            }
+
+            // validate request
+            // http://www.twilio.com/docs/security-reliability/security
+            // Take the full URL of the request, from the protocol (http...) through the end of the query string (everything after the ?)
+            var value = new StringBuilder();
+            var fullUrl = context.Request.Url.AbsoluteUri;
+
+            value.Append(fullUrl);
+
+            // If the request is a POST, take all of the POST parameters and sort them alphabetically.
+            if (context.Request.HttpMethod == "POST")
+            {
+                // Iterate through that sorted list of POST parameters, and append the variable name and value (with no delimiters) to the end of the URL string
+                var sortedKeys = context.Request.Form.AllKeys.OrderBy(k => k, StringComparer.Ordinal).ToList();
```

Comment:
That's not how `Select` is intended to be used. The key idea on `Select` a.k.a. _map_ is that when you apply the transformation function to all the elements in the collection you'll end up with another collection with the same size. `Aggregate` a.k.a. _reduce_ will _fold_ the elements in the collection into something else. Take a look at the canonical example for `Aggregate`:

```csharp
var nums = { 1, 2, 3, 4 };
var result = nums.Aggregate((a, b) => b + a); // result will be 10 as you might expect.

// This could make more sense for this example, though.
var sum = nums.Sum();
```

Both expressions are equivalent.